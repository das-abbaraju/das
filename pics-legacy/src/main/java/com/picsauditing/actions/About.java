package com.picsauditing.actions;

import java.text.SimpleDateFormat;
import java.util.Date;

import com.picsauditing.access.Anonymous;
import com.picsauditing.access.UserAgentParser;

public class About extends PicsActionSupport {

	private String browserName;
	private String operatingSystem;
	private Date systemTime;

	private static final long serialVersionUID = 1L;

	@Override
	@Anonymous
	public String execute() {
		UserAgentParser userAgentParser = new UserAgentParser(getRequest().getHeader("User-Agent"));
		browserName = userAgentParser.getBrowserName();
		operatingSystem = userAgentParser.getBrowserOperatingSystem();

		return SUCCESS;
	}

	public String getBrowserName() {
		return browserName;
	}

	public String getOperatingSystem() {
		return operatingSystem;
	}

	public String getSystemTime() {
		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
		systemTime = new Date();

		return dateFormat.format(systemTime);
	}
}