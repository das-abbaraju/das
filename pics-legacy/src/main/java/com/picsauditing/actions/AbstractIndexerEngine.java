package com.picsauditing.actions;

import com.picsauditing.dao.IndexableDAO;
import com.picsauditing.jpa.entities.*;
import com.picsauditing.search.Database;
import com.picsauditing.util.IndexerEngine;
import com.picsauditing.util.ReflectUtil;
import com.picsauditing.util.Strings;
import org.springframework.beans.factory.annotation.Autowired;

import java.sql.SQLException;
import java.util.HashSet;
import java.util.Set;

public abstract class AbstractIndexerEngine implements IndexerEngine {

    @Autowired
	protected IndexableDAO indexableDao;

	protected Database db = new Database();

	private static Set<Class<? extends Indexable>> entries = new HashSet<Class<? extends Indexable>>();

	static {
		entries.add(Account.class);
		entries.add(User.class);
		entries.add(Employee.class);
		entries.add(Trade.class);
	}

	@Override
	public void updateIndex(int id, Class<? extends Indexable> clazz) {
		Set<Integer> saved = new HashSet<Integer>();
		saved.add(id);
		updateIndex(saved, clazz);
	}

	@Override
	public void updateIndex(Set<Integer> saved, Class<? extends Indexable> clazz) {
		if (saved.isEmpty())
			return;
		String updateIndexing = "UPDATE " + ReflectUtil.getTableName(clazz) + " SET needsIndexing=0 WHERE id IN ("
				+ Strings.implode(saved) + ")";
		try {
			db.executeUpdate(updateIndexing);
		} catch (SQLException e) {
			// TODO Auto-generated catch block
		}
	}

	public void setIndexableDao(IndexableDAO indexableDao) {
		this.indexableDao = indexableDao;
	}

	public Set<Class<? extends Indexable>> getEntries() {
		return entries;
	}

}
