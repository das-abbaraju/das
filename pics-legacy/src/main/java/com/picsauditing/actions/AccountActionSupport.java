package com.picsauditing.actions;

import com.picsauditing.access.OpPerms;
import com.picsauditing.dao.AccountDAO;
import com.picsauditing.dao.CountryDAO;
import com.picsauditing.dao.CountrySubdivisionDAO;
import com.picsauditing.dao.NoteDAO;
import com.picsauditing.jpa.entities.*;
import com.picsauditing.model.contractor.AccountContact;
import com.picsauditing.util.Strings;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.*;

@SuppressWarnings("serial")
public class AccountActionSupport extends PicsActionSupport {

	protected int id;
	// protected Account account;
	protected String subHeading = null;
	// TODO This local-copy optimzation is probably unnecessary, being redundant
	// with Hibernate's caching.
	protected List<Note> notes;
	protected NoteCategory noteCategory = NoteCategory.General;
	protected int questionId;

	@Autowired
	protected AccountDAO accountDAO;
	@Autowired
	private CountrySubdivisionDAO countrySubdivisionDAO;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getQuestionId() {
		return questionId;
	}

	public void setQuestionId(int questionId) {
		this.questionId = questionId;
	}

	/**
	 * Delete this method and property when we start i18n
	 * 
	 * @return
	 */
	public String getSubHeading() {
		return subHeading;
	}

	public void setSubHeading(String subHeading) {
		this.subHeading = subHeading;
	}

	public void setAccount(Account account) {
		this.account = account;
	}

	public Account getAccount() {
		if (account == null) {
			loadPermissions();
			account = accountDAO.find(permissions.getAccountId(), permissions.getAccountType());
		}
		return account;
	}

	/************* NOTES ************/
	@Deprecated
	protected NoteDAO getNoteDao() {
		return noteDao;
	}

	public NoteCategory getNoteCategory() {
		return noteCategory;
	}

	public void setNoteCategory(NoteCategory noteCategory) {
		this.noteCategory = noteCategory;
	}

	/**
	 * Get a list of notes up to the limit, using the given where clause
	 * 
	 * @param where
	 *            should be in the format of "AND field=1", can be an empty
	 *            string
	 * @param limit
	 *            e.g. 25
	 * @return
	 */
	public List<Note> getNotes(String where, int firstLimit, int limit) {
		if (notes == null) {
			notes = noteDao.getNotes(id, permissions, "status IN (1,2)" + where, firstLimit, limit);
		}

		return notes;
	}

	/**
	 * Using the same parameters as getNotes(), above, see if there is at least
	 * one more note that can be fetched after that.
	 */
	public boolean atLeastOneMoreNote(String where, int firstLimit, int limit) {
		List<Note> zeroOrOneMoreNote = noteDao.getNotes(id, permissions, "status IN (1,2)" + where, firstLimit + limit,
				1);
		return (zeroOrOneMoreNote.size() > 0);
	}

	/**
	 * Get a list of 5 embedded notes, based on noteCategory
	 * 
	 * @return
	 */
	public List<Note> getNotes() {
		return getNotes(" AND noteCategory IN ('" + noteCategory.toString() + "','General')", 0, 5);
	}

	protected Note addNote(Account account, String newNote) {
		return addNote(account, newNote, noteCategory);
	}

	protected Note addNote(Account account, String newNote, NoteCategory noteCategory) {
		return addNote(account, newNote, noteCategory, LowMedHigh.Low, true, Account.EVERYONE, null, null);
	}

	protected Note addNote(Account account, String newNote, NoteCategory noteCategory, int viewableBy) {
		return addNote(account, newNote, noteCategory, LowMedHigh.Low, true, viewableBy, null, null);
	}

	protected Note addNote(Account account, String newNote, NoteCategory noteCategory, LowMedHigh priority,
			boolean canContractorView, int viewableBy, User user) {
		return addNote(account, newNote, noteCategory, LowMedHigh.Low, true, viewableBy, user, null);
	}

    protected Note addNote(Account account, String summary, String newNote, NoteCategory category, LowMedHigh priority,
                           boolean canContractorView, int viewableBy) {
        Note note = new Note();
        note.setAccount(account);
        note.setAuditColumns(permissions);
        note.setSummary(summary);
        note.setBody(newNote);
        note.setPriority(priority);
        note.setNoteCategory(category);
        note.setViewableById(viewableBy);
        note.setCanContractorView(canContractorView);
        note.setStatus(NoteStatus.Closed);
        noteDao.save(note);
        return note;
    }

    protected Note addNote(Account account, String newNote, NoteCategory category, LowMedHigh priority,
			boolean canContractorView, int viewableBy, User user, Employee employee) {
		Note note = new Note();
		note.setAccount(account);
		note.setAuditColumns(permissions);
		note.setSummary(newNote);
		note.setPriority(priority);
		note.setNoteCategory(category);
		note.setViewableById(viewableBy);
		note.setCanContractorView(canContractorView);
		note.setStatus(NoteStatus.Closed);
		note.setEmployee(employee);
		noteDao.save(note);
		return note;
	}

	public int getViewableByAccount(Account account) {
		if (account != null) {
			return account.getId();
		}
		return Account.EVERYONE;
	}

	/***** END of NOTES *****/

	@Deprecated
	public CountryDAO getCountryDAO() {
		return countryDAO;
	}

	@Deprecated
	public CountrySubdivisionDAO getCountrySubdivisionDAO() {
		return countrySubdivisionDAO;
	}

	public CountrySubdivision getAccountCountrySubdivision() {
		CountrySubdivision value = null;
		Account account = accountDAO.find(id, "Contractor");
		if (account != null) {
			value = account.getCountrySubdivision();
		}
		return value;
	}

	public String getCountrySubdivisionString() {
		String value = "";
		Account account = accountDAO.find(id, "Contractor");
		if (account != null) {
			value = account.getCountrySubdivision().getSimpleName();
		}
		return value;
	}

	public List<Country> getCountryList() {
		List<Country> countryList = countryDAO.findAll();
		Collections.sort(countryList, Country.NAME_COMPARATOR);
		countryList.add(0, countryDAO.find("GB"));
		countryList.add(0, countryDAO.find("CA"));
		countryList.add(0, countryDAO.find("US"));

		return countryList;
	}

	public List<CountrySubdivision> getCountrySubdivisionList() {
		if (account == null) {
			return getCountrySubdivisionList(null);
		}

		return getCountrySubdivisionList(account.getCountry().getIsoCode());
	}

	public List<CountrySubdivision> getCountrySubdivisionList(String countries) {
		List<CountrySubdivision> result;
		if (countries == null) {
			result = countrySubdivisionDAO.findByCountries(Country.COUNTRIES_WITH_SUBDIVISIONS, false);
		} else {
			boolean negative = false;
			if (countries.startsWith("!")) {
				countries = countries.replace("!", "");
				negative = true;
			}

			List<String> validCountriesWithSubdivisions = new ArrayList<>();
			for (String country : countries.split("[|]")) {
				if (Country.COUNTRIES_WITH_SUBDIVISIONS.contains(country)) {
					validCountriesWithSubdivisions.add(country);
				}
			}

			result = countrySubdivisionDAO.findByCountries(validCountriesWithSubdivisions, negative);
		}

		Collections.sort(result, CountrySubdivision.NAME_COMPARATOR);

		return result;
	}

	public String getCountrySubdivisionLabelFor(String country) {
		return getText(getCountrySubdivisionLabelKeyFor(country));
	}

	public String getCountrySubdivisionLabelKeyFor(String country) {
		if (Strings.isNotEmpty(country)) {
			String countrySubdivisionLabelKey = String.format(Country.COUNTRY_SUBDIVISION_LABEL_FORMAT, country);

			if (hasKey(countrySubdivisionLabelKey)) {
				return countrySubdivisionLabelKey;
			}
		}

		return Country.DEFAULT_COUNTRY_SUBDIVISION_LABEL;
	}

	public boolean isShowMoreNotes() {
		if (Account.CONTRACTOR_ACCOUNT_TYPE.equals(account.getType())) {
			return permissions.hasPermission(OpPerms.ContractorDetails);
		}

		return true;
	}

	public AccountStatus[] getStatusList() {
		return AccountStatus.values();
	}

    public String getAccountLastLogin() {
        Date lastLogin = accountDAO.findLastAccountLogin(account.getId());
        return getFuzzyDate(lastLogin);
    }

    public List<AccountContact> getAccountContactsByRole(OpPerms opPerms) {
        List<AccountContact> results = accountDAO.findAccountContactsByRole(account.getId(), opPerms);
        return results;
    }
}
