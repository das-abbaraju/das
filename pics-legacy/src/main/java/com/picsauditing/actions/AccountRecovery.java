package com.picsauditing.actions;

import com.picsauditing.PICS.Grepper;
import com.picsauditing.access.Anonymous;
import com.picsauditing.authentication.entities.AppUser;
import com.picsauditing.authentication.service.AppUserService;
import com.picsauditing.dao.UserDAO;
import com.picsauditing.employeeguard.entities.Profile;
import com.picsauditing.employeeguard.services.entity.ProfileEntityService;
import com.picsauditing.jpa.entities.User;
import com.picsauditing.mail.EmailBuildErrorException;
import com.picsauditing.service.email.AccountRecoveryEmailService;
import com.picsauditing.util.Strings;
import com.picsauditing.util.URLUtils;
import com.picsauditing.validator.InputValidator;
import org.springframework.beans.factory.annotation.Autowired;

import java.io.IOException;
import java.util.Date;
import java.util.List;

@SuppressWarnings("serial")
public class AccountRecovery extends PicsActionSupport {
	@Autowired
	private UserDAO userDAO;
    @Autowired
    private AppUserService appUserService;
    @Autowired
    private ProfileEntityService profileService;
	@Autowired
	private InputValidator inputValidator;
    @Autowired
    private AccountRecoveryEmailService emails;

	private String email, username;
	private URLUtils urlUtils;

    private AccountRecoveryMessages messages = new AccountRecoveryMessages(this);

    private static final String PASSWORD = "password";
    private static final String USERNAME = "username";

    private static final Grepper<User> recoverableUserFilter = new Grepper<User>() {
        @Override
        public boolean check(User user) {
            return user != null && !user.isDeleted();
        }
    };

	@Anonymous
	@Override
	public String execute() throws Exception {
		return PASSWORD;
	}

	@Anonymous
	public String recoverUsername() {
		return USERNAME;
	}

	@Anonymous
	public String findName() throws Exception {

        if (!inputValidator.validateEmail(email).equals(InputValidator.NO_ERROR)) {
            messages.addNoUserNameError();
			return USERNAME;
		}

		final List<User> matchingUsers = recoverableUserFilter.grep(userDAO.findByEmail(email.trim()));

		if (matchingUsers.isEmpty()) {
            messages.addUserEmailNotFoundError();
            return accountRecoveryRedirect();
		} else {
            try {
                emails.sendUsernameRecoveryEmail(matchingUsers);
                messages.addUsernameEmailSentMessage();
                return loginRedirect();
            } catch (Exception e) {
                messages.addNoEmailSentError();
                return USERNAME;
            }
        }
	}



    private String loginRedirect() throws IOException {
        String loginPage = urlUtils().getActionUrl("Login");
        return setUrlForRedirect(loginPage);
    }

    private String accountRecoveryRedirect() throws IOException {
        String accountRecoveryPage = urlUtils().getActionUrl("AccountRecovery", "recoverUsername");
        return setUrlForRedirect(accountRecoveryPage);
    }

	@Anonymous
	public String resetPassword() {
		if (Strings.isEmpty(username) || username.startsWith("DELETE-")) {
            messages.addBlankUsernameSubmittedError();
			return PASSWORD;
		}

        final AppUser appUser = appUserService.findByUsername(username);

        if (appUser == null) {
            messages.addUsernameNotFoundError();
        } else {
            final User user = userDAO.findName(username);
            final Profile profile = profileService.findByAppUserId(appUser.getId());

            if (user == null && profile == null) {
                messages.addNoValidAccountFound();
            } else if (!user.isActiveB() && profile == null) {
                messages.addUserNotActiveError();
            } else {
                try {
                    addResetHashTo(appUser);
                    appUserService.save(appUser);

                    if (user != null) {
                        emails.sendRecoveryEmail(user, getRequestHost());
                        messages.successfulEmailSendTo(user.getEmail());
                    } else {
                        emails.sendRecoveryEmail(profile, getRequestHost());
                        messages.successfulEmailSendTo(profile.getEmail());
                    }

                    return setUrlForRedirect("Login.action");

                } catch (EmailBuildErrorException e) {
                    messages.failedEmailSend();
                } catch (IOException e) {
                    messages.addUsernameNotFoundError();
                }
            }
        }
        return PASSWORD;
	}

    private void addResetHashTo(AppUser appUser) {
        appUser.setResetHash(Strings.hashUrlSafe("u" + appUser.getId() + String.valueOf(new Date().getTime())));
    }

    public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}


	private URLUtils urlUtils() {
		if (urlUtils == null) {
			urlUtils = new URLUtils();
		}
		return urlUtils;
	}


}