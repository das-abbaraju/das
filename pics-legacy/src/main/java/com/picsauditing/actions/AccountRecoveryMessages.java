package com.picsauditing.actions;

import com.picsauditing.jpa.entities.User;

class AccountRecoveryMessages {

    static final String EMAIL_SENT_HEADER_KEY = "AccountRecovery.EmailSentHeader";
    static final String RECOVERY_EMAIL_SENT_KEY = "AccountRecovery.EmailSent";
    static final String USERNAME_EMAIL_SENT_KEY = "AccountRecovery.EmailSent2";
    static final String EMAIL_RESET_ERROR_KEY = "AccountRecovery.error.ResetEmailError";
    static final String NO_USERNAME_ERROR_KEY = "AccountRecovery.error.NoUserName";
    static final String EMAIL_NOT_FOUND_KEY = "AccountRecovery.error.EmailNotFound";
    static final String NO_EMAIL_SENT_ERROR_KEY = "AccountRecovery.error.NoEmailSent";
    static final String BLANK_USERNAME_INPUT_ERROR_KEY = "AccountRecovery.error.NoUserName";
    static final String USER_NOT_FOUND_ERROR_KEY = "AccountRecovery.error.UserNotFound";
    static final String USER_NOT_ACTIVE_ERROR_KEY = "AccountRecovery.error.UserNotActive";
    static final String USERNAME_NOT_FOUND_ERROR_KEY = "AccountRecovery.error.UserNameNotFound";

    final private PicsActionSupport parent;

    AccountRecoveryMessages(PicsActionSupport parent) {
        this.parent = parent;
    }

    private void message(String messageKey) {
        parent.addActionMessage(parent.getText(messageKey));
    }

    private void error(String messageKey) {
        parent.addActionError(parent.getText(messageKey));
    }

    private void setHeader() {
        parent.setActionMessageHeader(parent.getText(EMAIL_SENT_HEADER_KEY));
    }

    private void messageWithSentHeader(String key) {
        setHeader();
        message(key);
    }

    void addNoUserNameError() { error(NO_USERNAME_ERROR_KEY); }
    void addUserNotFoundError() { error(USER_NOT_FOUND_ERROR_KEY); }
    void addUsernameNotFoundError() { error(USERNAME_NOT_FOUND_ERROR_KEY); }
    void addUserEmailNotFoundError() { error(EMAIL_NOT_FOUND_KEY); }
    void addUserNotActiveError() { error(USER_NOT_ACTIVE_ERROR_KEY); }
    void addNoEmailSentError() { error(NO_EMAIL_SENT_ERROR_KEY); }
    void addBlankUsernameSubmittedError() { error(BLANK_USERNAME_INPUT_ERROR_KEY); }
    void addUsernameEmailSentMessage() { messageWithSentHeader(USERNAME_EMAIL_SENT_KEY); }

    void failedEmailSend() { error(EMAIL_RESET_ERROR_KEY); }
    void successfulEmailSendTo(User user) {
        setHeader();
        parent.addActionMessage(parent.getTextParameterized(RECOVERY_EMAIL_SENT_KEY, user.getEmail()));
    }

}
