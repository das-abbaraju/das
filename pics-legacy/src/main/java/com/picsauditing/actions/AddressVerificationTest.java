package com.picsauditing.actions;

import com.picsauditing.service.addressverifier.AddressHolder;
import com.picsauditing.service.addressverifier.AddressVerificationService;
import org.springframework.beans.factory.annotation.Autowired;

public class AddressVerificationTest extends PicsActionSupport {
    @Autowired
    private AddressVerificationService addressVerificationService;

    private AddressHolder address;

    public String execute() throws Exception {

        if (address != null) {
            address = addressVerificationService.verify(address);
        }

        return SUCCESS;
    }

    public AddressHolder getAddress() {
        return address;
    }

    public void setAddress(AddressHolder address) {
        this.address = address;
    }
}
