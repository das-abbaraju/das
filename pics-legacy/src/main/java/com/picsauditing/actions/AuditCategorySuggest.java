package com.picsauditing.actions;

import java.util.ArrayList;
import java.util.List;

import com.picsauditing.dao.AuditCategoryDAO;
import com.picsauditing.jpa.entities.AuditCategory;

@SuppressWarnings("serial")
public class AuditCategorySuggest extends PicsActionSupport {
	private String q;
	private List<AuditCategory> results;
	private AuditCategoryDAO auditCategoryDAO;

	public AuditCategorySuggest(AuditCategoryDAO auditCategoryDAO) {
		this.auditCategoryDAO = auditCategoryDAO;
	}

	public String execute() throws Exception {
		results = new ArrayList<AuditCategory>();

		if (q == null || q.length() < 3)
			return SUCCESS;

		results = auditCategoryDAO.findCategoryNames(q);

		if (results == null)
			results = new ArrayList<AuditCategory>();

		return SUCCESS;
	}

	public String getQ() {
		return q;
	}

	public void setQ(String search) {
		this.q = search;
	}

	public List<AuditCategory> getResults() {
		return results;
	}

	public void setResults(List<AuditCategory> results) {
		this.results = results;
	}

}
