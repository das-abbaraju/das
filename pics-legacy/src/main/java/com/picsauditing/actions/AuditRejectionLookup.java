package com.picsauditing.actions;

import java.util.List;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.util.CollectionUtils;

import com.picsauditing.dao.AuditRejectionCodeDAO;
import com.picsauditing.dao.ContractorAuditOperatorDAO;
import com.picsauditing.jpa.entities.AuditRejectionCode;
import com.picsauditing.jpa.entities.ContractorAuditOperator;

@SuppressWarnings("serial")
public class AuditRejectionLookup extends PicsActionSupport {

	@Autowired
	private AuditRejectionCodeDAO auditRejectionCodeDao;
	@Autowired
	private ContractorAuditOperatorDAO contractorAuditOperatorDao;
	
	private int caoId;

	@Override
	public String execute() throws Exception {
		ContractorAuditOperator cao = contractorAuditOperatorDao.find(caoId);
		if (cao != null) {
			List<AuditRejectionCode> auditRejectionCodes = auditRejectionCodeDao.findByCaoPermissions(cao.getCaoPermissions());
			jsonArray = populateJsonArray(auditRejectionCodes);
		}
		
		return SUCCESS;
	}
	
	@SuppressWarnings("unchecked")
	private JSONArray populateJsonArray(List<AuditRejectionCode> auditRejectionCodes) {
		JSONArray jsonArray = new JSONArray();
		if (CollectionUtils.isEmpty(auditRejectionCodes)) {
			return jsonArray;
		}
		
		for (AuditRejectionCode auditRejectionCode : auditRejectionCodes) {
			JSONObject jsonObject = new JSONObject();
			jsonObject.put("id", auditRejectionCode.getAuditSubStatus().toString());
			jsonObject.put("value", auditRejectionCode.getRejectionReason());
			jsonArray.add(jsonObject);
		}
		
		return jsonArray;
	}
	
	public int getCaoId() {
		return caoId;
	}
	
	public void setCaoId(int caoId) {
		this.caoId = caoId;
	}

}
