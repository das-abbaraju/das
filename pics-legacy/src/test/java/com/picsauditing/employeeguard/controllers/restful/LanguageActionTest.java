package com.picsauditing.employeeguard.controllers.restful;

import com.picsauditing.PicsActionTest;
import com.picsauditing.actions.PicsActionSupport;
import com.picsauditing.model.i18n.KeyValue;
import com.picsauditing.model.i18n.LanguageModel;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;

import java.util.ArrayList;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.when;

@UseReporter(DiffReporter.class)
public class LanguageActionTest extends PicsActionTest {

	@Mock
	private LanguageModel languageModel;

	// Class under test
	private LanguageAction languageAction;

	@Before
	public void setUp() throws Exception {
		MockitoAnnotations.initMocks(this);

		languageAction = new LanguageAction();

		super.setUp(languageAction);

		Whitebox.setInternalState(languageAction, languageModel);
	}

	@Test
	public void testIndex() throws Exception {
		setupTestIndex();

		String result = languageAction.index();

		verifyTestIndex(result);
	}

	private void setupTestIndex() {
		when(languageModel.getVisibleLanguagesSansDialect()).thenReturn(new ArrayList<KeyValue<String, String>>() {{

			add(new KeyValue<>("en", "English"));
			add(new KeyValue<>("de", "Deutsch"));
			add(new KeyValue<>("nl", "Nederlands"));

		}});
	}

	private void verifyTestIndex(String result) throws Exception {
		assertEquals(PicsActionSupport.JSON_STRING, result);

		Approvals.verify(languageAction.getJsonString());
	}
}
