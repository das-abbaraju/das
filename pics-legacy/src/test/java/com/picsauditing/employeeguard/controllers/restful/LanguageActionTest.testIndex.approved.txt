[{"id":"en","name":"English"},{"id":"de","name":"Deutsch"},{"id":"nl","name":"Nederlands"}]